assignment-0/controller.go

```diff
@@ -0,0 +1,113 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+)
+
+type controller struct {
+	Client   *kubernetes.Clientset
+	Informer cache.SharedIndexInformer
+}
+
+func newController(client *kubernetes.Clientset, informerFactory informers.SharedInformerFactory) *controller {
+	deploymentInformer := informerFactory.Apps().V1().Deployments().Informer()
+	deploymentInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			createService(client, obj)
+		},
+		DeleteFunc: func(obj interface{}) {
+			deleteService(client, obj)
+		},
+	})
+
+	return &controller{
+		Client:   client,
+		Informer: deploymentInformer,
+	}
+}
+
+func (c *controller) run(ch <-chan struct{}) {
+	fmt.Println("starting controller")
+	if !cache.WaitForCacheSync(ch, c.Informer.HasSynced) {
+		fmt.Print("waiting for cache to be synced\n")
+	}
+
+	go wait.Until(c.worker, 1*time.Second, ch)
+
+	<-ch
+}
+
+func (c *controller) worker() {
+	for c.processItem() {
+	}
+}
+
+func (c *controller) processItem() bool {
+	return true
+}
```

Comment:
Do we really need this if we are just returning `true` always.

Reply:
modified this function